include/cglm/types.h

```diff
@@ -32,6 +32,16 @@
 #  define CGLM_ALIGN_MAT CGLM_ALIGN(16)
 #endif
 
+#ifdef __GNUC__
+#  define CGLM_ASSUME_ALIGNED(expr, alignment) \
+  __builtin_assume_aligned((expr), (alignment))
+#else
+#  define CGLM_ASSUME_ALIGNED(expr, alignment) (expr)
+#endif
+
+#define CGLM_CASTPTR_ASSUME_ALIGNED(expr, type) \
+  ((type*)CGLM_ASSUME_ALIGNED((expr), __alignof__(type)))
```

Comment:
I think `__alignof__` may not be standard, in this case not all compilers may support that. It would be nice to see that **cglm** works on all compilers ( including msvc ). Maybe we can check compiler support for that?

Reply:
`alignof()` is standard since C11, not sure about `__alignof__()` or MSVC's status other than `alignof()` being supported there since C++11 (maybe they've had it in for C, too?)

Reply:
This doesn't matter because the second argument of the `CGLM_ASSUME_ALIGNED` macro is only expanded for the `__GNUC__` implementation. GCC (and all compilers that pretend to be GCC) is expected to have `__alignof__` in all `-std=` modes (unlike `alignof`, which may not be available in strict modes).

MSVC and the rest will get this implementation
```c
#  define CGLM_ASSUME_ALIGNED(expr, alignment) (expr)
```
Which just ignores the `alignment` argument as you can see.

Reply:
> alignof() is standard since C11

Good to know

> MSVC and the rest will get this implementation

Sorry, I missed the flow.